pkg/sql/schemachanger/scbuild/internal/scbuildstmt/dependencies.go

```diff
@@ -325,3 +358,15 @@ type NameResolver interface {
 	// ResolveConstraint retrieves a constraint by name and returns its elements.
 	ResolveConstraint(relationID catid.DescID, constraintName tree.Name, p ResolveParams) ElementResultSet
 }
+
+// ReferenceProvider provides all referenced objects with in current DDL
+// statement. For example, CREATE VIEW and CREATE FUNCTION both could reference
+// other objects, and cross-references need to probably tracked.
+type ReferenceProvider interface {
+	// ForEachTableReference iterate through all referenced tables and the
+	// reference details with the given function.
+	ForEachTableReference(func(tblID descpb.ID, refs []descpb.TableDescriptor_Reference) error) error
```

Comment:
Done. Thanks for the suggestion. I made the interface into more detailed methods so that we don't need to reason about things like descriptor types (view, sequence, type, table) anymore.